protos/merlin/transformer/spec/encoder.proto

```diff
@@ -0,0 +1,25 @@
+syntax = "proto3";
+
+package merlin.transformer;
+
+option go_package = "github.com/gojek/merlin/pkg/transformer/spec";
+
+message Encoder {
+  string name = 1;
+  oneof encoderConfig {
```

Comment:
I think Encoder message still need an `encoderType` field here to store the encoderConfig type value. There are three reasoning behind this:
1. If user accidentally (or intentionally) specify two encoderConfigs, we can refer to `encoderType` to decide which one to use.
2. It helps UI component to preserve the other config. For example, I created and configured ordinal encoder, then I switch to other encoder type. But when I want to use ordinal encoder again, the config is still there.
3. In the API implementation, we can refer to `encoderType` to do casting instead of checking if the config is nil or not.

Reply:
I prefer not to add this because there is not much value by adding `encoderType` cause it is will be redundant with the config, and also prone to make mistake if the config specified is mismatch with the type (if user use sdk)

* If user accidentally specify two encoderConfigs, it will choose the latest one specified based on the proto `oneOf` behaviour, and `encoderType` can't decide which one to use.
* I think UI can still achieve that behaviour without encoderType, or with encoderType which the type won't be processed in the backend since it only need the config 
* there won't be issue with config is nil IMO, because it will failed in deployment (because invalid proto)